workflow/gccontroller/gc_controller.go

```diff
@@ -1,44 +1,55 @@
-package ttlcontroller
+package gccontroller
 
 import (
 	"context"
 	"fmt"
+	"sort"
 	"time"
 
 	log "github.com/sirupsen/logrus"
 	apierr "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/clock"
 	runtimeutil "k8s.io/apimachinery/pkg/util/runtime"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/util/workqueue"
 
+	"github.com/argoproj/argo-workflows/v3/config"
 	wfv1 "github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1"
 	wfclientset "github.com/argoproj/argo-workflows/v3/pkg/client/clientset/versioned"
 	commonutil "github.com/argoproj/argo-workflows/v3/util"
+	envutil "github.com/argoproj/argo-workflows/v3/util/env"
 	"github.com/argoproj/argo-workflows/v3/workflow/common"
+	"github.com/argoproj/argo-workflows/v3/workflow/controller/indexes"
 	"github.com/argoproj/argo-workflows/v3/workflow/metrics"
 	"github.com/argoproj/argo-workflows/v3/workflow/util"
 )
 
 type Controller struct {
-	wfclientset wfclientset.Interface
-	wfInformer  cache.SharedIndexInformer
-	workqueue   workqueue.DelayingInterface
-	clock       clock.Clock
-	metrics     *metrics.Metrics
+	wfclientset       wfclientset.Interface
+	wfInformer        cache.SharedIndexInformer // Maybe this is the correct index but probably not.
```

Comment:
what does this comment mean?

Reply:
Misstake from me to leave it here. Left it as a note for myself when I had to read up on things.